catalog/security/banner-marking/src/main/java/org/codice/alliance/security/banner/marking/BannerCommonMarkingExtractor.java

```diff
@@ -81,17 +78,23 @@ private void checkSecurityAttributes(String name, List<Serializable> oldVals,
             return;
         }
 
-        Set<String> oldValues = new HashSet<>((List<String>) (List<?>) oldVals);
-        Set<String> newValues = new HashSet<>((List<String>) (List<?>) newVals);
-        if (!oldValues.equals(newValues)) {
-            String errorMessage = String.format(
-                    "Extracted security attribute %s from the banner markings"
-                            + " does not match value pre-existing on the metacard. The"
-                            + " metacard will not be created.",
-                    name);
-            INGEST_LOGGER.error(errorMessage);
-            throw new RuntimeException(errorMessage);
-        }
+        //assert lists are the same
+        oldVals.forEach(val -> {
+            if (!newVals.contains(val)) {
+                throwSecurityMismatchException(name);
+            }
+        });
+        newVals.forEach(val -> {
```

Comment:
❗️ Instead of looping over both, just check that the sizes match first and then if they do, loop over one or the other.

Reply:
Oh nice idea!